framework/src/media/audio/audio_manager.c

```diff
@@ -453,22 +454,23 @@ static audio_manager_result_t get_audio_volume(int fd, audio_config_t *config, a
 		caps_desc.caps.ac_format.hw = AUDIO_FU_INP_GAIN;
 	}
 
-	ret = ioctl(fd, AUDIOIOC_GETCAPS, (unsigned long *)&caps_desc.caps);
-	if (ret < 0) {
-		meddbg("Fail to ioctl AUDIOIOC_GETCAPS\n");
-		return AUDIO_MANAGER_DEVICE_FAIL;
+	if ((ioctl(fd, AUDIOIOC_GETCAPS, (unsigned long *)&caps_desc.caps)) < 0) {
+		// Default gain/volume value is provided even though CONFIG_AUDIO_EXCLUDE_VOLUME is defined.
+		config->excluded = true;
+		ret = AUDIO_MANAGER_DEVICE_NOT_SUPPORT;
+		meddbg("ret = %d, Volume not supported, excluded = %d\n", ret, config->excluded);
 	}
-
 	max_volume = caps_desc.caps.ac_controls.hw[0];
 	cur_volume = caps_desc.caps.ac_controls.hw[1];
 
 	/* scale here */
+	ASSERT(max_volume > 0);
```

Comment:
This is framework, not platform so assert is not needed, I think.

Reply:
Deleted.